usmqe_tests/conftest.py

```diff
@@ -374,24 +374,35 @@ def valid_session_credentials(request):
 @pytest.fixture
 def cluster_reuse(valid_session_credentials):
     """
-    Returns cluster identified by one of machines
-    from cluster.
+    Returns Gluster *trusted storage pool* (aka cluster) as identified by
+    ``cluster_member`` machine (specified in usmqe config file).
+
     Returned cluster can be used for further testing.
     Function uses Tendrl API(clusters). In case there
     is need to identify cluster directly by storage
     tools this function should be split.
     """
+    LOGGER = pytest.get_logger("cluster_reuse")
     id_hostname = CONF.config["usmqe"]["cluster_member"]
     api = TendrlApi(auth=valid_session_credentials)
     for _ in range(12):
-        clusters = api.get_cluster_list()
-        clusters = [cluster for cluster in clusters
-                    if id_hostname in
-                    [node["fqdn"] for node in cluster["nodes"]]
-                    ]
+        clusters = []
+        for cluster in api.get_cluster_list():
+            node_fqdn_list = [node["fqdn"] for node in cluster["nodes"]]
+            if id_hostname in node_fqdn_list:
+                clusters.append(cluster)
+                LOGGER.debug("found cluster for cluster member {}".format(id_hostname))
+            else:
+                msg = "cluster member {} not found in cluster {}"
+                LOGGER.debug(msg.format(id_hostname, node_fqdn_list))
         if len(clusters) == 1:
-            return clusters[0]
+            cluster = clusters[0]
+            LOGGER.info("using: {}".format([node["fqdn"] for node in cluster["nodes"]]))
+            return cluster
+        else:
+            LOGGER.warning("unexpected number (!= 1) of clusters found: {}".format(len(clusters)))
```

Comment:
What about print here also the content of `clusters` variable? If it is empty, it is not needed, but what if there will be for some reason 2 or more clusters?... then it might be helpful for debugging. But it is true, that it should be possible to find this info from the previous debug messages...

Reply:
Good point.